### PR TITLE
added setAnonymize IP

### LINKED ISSUE #119: setAnonymizeIp for privacy issues
I need to be able to set anonymizeIP to true for privacy concerns can a method or flag be added for this functionality?

https://developers.google.com/android/reference/com/google/android/gms/analytics/Tracker#setAnonymizeIp(boolean)

### PR DESCRIPTION
**Pull request by Freundschaft**
this should fix #119 and #83

### CODE REVIEW

**Comment by wenzerl** [comment]
Hi,
any news on this one? Will these commits be merged?
Thanks, Werner

**Comment by ma-pe** [comment]
+1

**Comment by blabno** [comment]
How about resolving concflicts?

**Comment by Freundschaft** [comment]
phew, this PR is quite a while ago, I can look into it, but probably only will be able to do so next week

**Comment by wenzerl** [comment]
that would be totally awesome!

**Comment by wenzerl** [comment]
Any news on this one?
I would have absolutely no problem, if there is an financial issue. So i am more then willing to pay for that change to be made. Please let me know if this is an option. Cheers Werner

**Comment by Freundschaft** [comment]
sorry, I have been extremely busy so far in january, hopefullly will be able to do something next week.

**Comment by wenzerl** [comment]
that would be totally awesome!

**Comment by Freundschaft** [comment]
not sure if this works but someone is free to pull and check if setanonymizeip actually works, at least the plugin runs right now

**Comment by wenzerl** [comment]
Great, Freundschaft!
But how does this now work? Should Dan Wilson merge those branches?
How can i install this new version?

**Comment by wenzerl** [comment]
Hi Freundschaft,
we just checkout your commit. The question now is, how to activate IP-Anonymize. Is there something to setup?

**Comment by Freundschaft** [comment]
this should work, but as I said I havent verified yet if anonymizeIP is actually activated by the changes.

```
if ('analytics' in $window) {
                Analytics.startTrackerWithId(GOOGLE_ANALYTICS_ID, function () {
                    Analytics.setAnonymizeIp(true, function () {
                        //Analytics.debugMode(function(){
                        $log.info('Google Analytics Plugin loaded');
                        //}, $log.error)
                    }, $log.error)
                }, $log.error);
            } else {
                $log.warn('Google Analytics unavailable (only available on the device since it\'s a native cordova plugin');
}
```

**Comment by wenzerl** [comment]
Thanks for the effort!!
We now decided to use that one:
https://github.com/cmackay/google-analytics-plugin
and follow his instructions from https://github.com/cmackay/google-analytics-plugin/issues/43

**Comment by Freundschaft** [comment]
no worries. Do you know how to verify if setting anonymize ip actually worked or not?

**Comment by wenzerl** [comment]
No, i am just hoping it does work... No idea how to check that.

**Comment by Freundschaft** [comment]
:+1:

**Comment by victorsosa** [comment]
Thank you, you are awesome; please update the readme file so I can merge this change.

**Comment by victorsosa** [comment]
@Freundschaft please update branch to fix the conflict

**Comment by Freundschaft** [comment]
alright sorry, will check next week, this week I'm still packed with work :(

**Comment by victorsosa** [comment]
No problema, thank you

**Comment by MrJH** [comment]
+1

**Comment by Freundschaft** [comment]
I fixed the merge conflicts but didnt test functionality yet. Can someone test? Otherwise I'll test it today or tomorrow and confirm that its ready for merge

**Comment by victorsosa** [comment]
Thanks please add he method to the typescript definition file too - analytics.d.ts; I will wait until this is tested, thanks.

**Comment by Freundschaft** [comment]
alright just checked and seems to work at least on android:

```
07-15 00:43:40.903 I/GAv4    (20294):   adb shell setprop log.tag.GAv4 DEBUG
07-15 00:43:40.903 I/GAv4    (20294):   adb logcat -s GAv4
07-15 00:43:40.916 D/GAv4    (20294): setLocalDispatchPeriod (sec): 30
07-15 00:43:52.155 D/GAv4    (20294): Hit delivery requested: ht=xxxxxxxxxxxxx,
_s=62, _v=ma9.2.56, a=xxxxxxxxxx, aid=com.gfnork.roadAtlas, aip=1, an=RoadAtlas,
av=0.2.50, cd=/overview/:state, cid=xxxxxxxxxxxxxxxxxxx, sr=108
0x1920, t=screenview, tid=xxxxxxxxx, ul=en-us, v=1
07-15 00:43:52.194 D/GAv4    (20294): Hit sent to the device AnalyticsService fo
r delivery
```

the aip=1 confirms that anonymize ip is enabled according to https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#aip

Cheers

File: typescript/analytics.d.ts [hunk 1]
```diff
@@ -1,44 +1,47 @@
...
 	public startTrackerWithId(id:String, successCallback?:Function, errorCallback?:Function):void;
-	
+
 	/** Sets a UserId */
 	public setUserId(id:String, successCallback?:Function, errorCallback?:Function):void;
-	
-	/** Sets a AppVersion */
+
+  /** Sets a setAnonymizeIp */
+  public setAnonymizeIp(anonymize:String, successCallback?:Function, errorCallback?:Function):void;
```
Comment (victorsosa) [inline typescript/analytics.d.ts]: anonymize:String should be boolean the parameter

Comment (Freundschaft) [inline typescript/analytics.d.ts]: just fixed it, thanks

**Comment by victorsosa** [comment]
Awesome :+1:

**Pull request merged**